devdoc/singlylinkedlist2_requirements.md

```diff
@@ -0,0 +1,176 @@
+singlylinkedlist2 requirements
+================
+
+## Overview
+
+SinglyLinkedList is module that provides the functionality of a singly linked list, allowing its user to add, remove and iterate the list elements.
+
+## Exposed API
+
+```c
+typedef struct SLIST_ENTRY_TAG
+{
+    struct SLIST_ENTRY_TAG *next;
+} SLIST_ENTRY, *PSLIST_ENTRY;
+
+typedef bool (*SLIST_MATCH_FUNCTION)(PSLIST_ENTRY listEntry, void* matchContext);
+typedef bool (*SLIST_CONDITION_FUNCTION)(PSLIST_ENTRY listEntry, void* matchContext, bool* continueProcessing);
+typedef int (*SLIST_ACTION_FUNCTION)(PSLIST_ENTRY listEntry, void* action_context, bool* continue_processing);
```

Comment:
>int [](http://example.com/codeflow?start=8&length=3)

I think we could have a void function here (or have the function return whether to continue processing or not).

Reply:
Unless we're diverging from what Dlist is doing. I don't have the Dlist code in front of me.

Reply:
DList action function return an int, should I keep the same or change it to void?

Reply:
Ok, let's keep it like DLIST